src/DeviceStatusPage/IntrospectionCard.tsx

```diff
@@ -31,28 +32,36 @@ interface IntrospectionTableProps {
 const IntrospectionTable = ({
   deviceId,
   introspection,
-}: IntrospectionTableProps): React.ReactElement => (
-  <Table responsive>
-    <thead>
-      <tr>
-        <th>Name</th>
-        <th>Major</th>
-        <th>Minor</th>
-      </tr>
-    </thead>
-    <tbody>
-      {introspection.map((iface) => (
-        <tr key={iface.name}>
-          <td>
-            <Link to={`/devices/${deviceId}/interfaces/${iface.name}`}>{iface.name}</Link>
-          </td>
-          <td>{iface.major}</td>
-          <td>{iface.minor}</td>
+}: IntrospectionTableProps): React.ReactElement => {
+  const astarte = useAstarte();
+
+  return (
+    <Table responsive>
+      <thead>
+        <tr>
+          <th>Name</th>
+          <th>Major</th>
+          <th>Minor</th>
         </tr>
-      ))}
-    </tbody>
-  </Table>
-);
+      </thead>
+      <tbody>
+        {introspection.map((iface) => (
+          <tr key={iface.name}>
+            <td>
+              {astarte.token?.can('realmManagement', 'GET', '/interfaces') ? (
```

Comment:
I would be more specific here, checking whether the token allows to GET values from that interface before accessing the interface page.

```suggestion
              {astarte.token?.can('realmManagement', 'GET', `/devices/${device_id}/interfaces/${iface.name}`) ? (
```

It would be nice to add similar checks in the rest of the UI to avoid showing buttons, links or editable fields where the action is not permitted by the token.